### PR TITLE
Continue migrating HTML form fields to XML fields.

### PR DESCRIPTION
**Pull request by joesoftware**
No "capitalization" or "rounding" in this PR, as I wanted to migrate as many text fields as possible.
None of the radio-button or Y/N fields are migrated.
Don't know what to do about the XML nodes `VetHealthInsureCountry` and `VetHealthInsurePostalCode`, as I didn't know which (if any) HTML fields to migrate from.
Several date fields that are combined on the form need to be split for XML.
Need to take turn `preferred_va_facility` into state/address/etc fields in XML.

### CODE REVIEW

File: _health-care/_js/_form.js [hunk 1]
```diff
@@ -41,6 +41,41 @@ function extractMonetaryValueTransform(formData, arg) {
...
+
+function extractAddressStateTransform(formData, arg) {
+  return extractTransform(formData, arg + "[state]");
+}
+
+function extractAddressZipcodeTransform(formData, arg) {
+  return extractTransform(formData, arg + "[zipcode]");
+}
+
+function extractAddressPhoneTransform(formData, arg) {
```
Comment (akainic) [inline _health-care/_js/_form.js]: Is this supposed to be called `extractPhoneTransform` instead?

File: _health-care/_js/_form.js [hunk 2]
```diff
@@ -41,6 +41,41 @@ function extractMonetaryValueTransform(formData, arg) {
...
+  return extractTransform(formData, arg + "[phone]");
+}
+
+function extractAddressWithFormatTransform(formData, arg) {
+  let name = extractAddressNameTransform(formData, arg);
+  let street = extractAddressStreetTransform(formData, arg);
+  let city = extractAddressCityTransform(formData, arg);
+  let state = extractAddressStateTransform(formData, arg);
+  let zip = extractAddressZipcodeTransform(formData, arg);
+  let phone = extractPhoneTransform(formData, arg);
```
Comment (akainic) [inline _health-care/_js/_form.js]: Or maybe this should be `extractAddressPhoneTransform`? I think you're getting one JS test failure due to this discrepancy, but then when I run it locally I get a different error after that...

File: _health-care/_js/_form.js [hunk 3]
```diff
@@ -1056,37 +1095,45 @@ let xmlFieldMap = [
...
   { node: "SpouseDateOfBirthDay",  name: 'veteran[]', arg: "3", transform: identityTransform },
   { node: "SpouseDateOfBirthYear",  name: 'veteran[]', arg: "1990", transform: identityTransform },
   { node: "MarraigeDateMonth",  name: 'veteran[]', arg: "6", transform: identityTransform },
   { node: "MarraigeDateDay",  name: 'veteran[]', arg: "8", transform: identityTransform },
   { node: "MarraigeDateYear",  name: 'veteran[]', arg: "2010", transform: identityTransform },
-  { node: "SpouseAddress",  name: 'veteran[]', arg: "1 Spouse Dr", transform: identityTransform },
-  { node: "SpouseCity",  name: 'veteran[]', arg: "Auburn", transform: identityTransform },
+  { node: "SpouseAddress",  arg: 'veteran[spouses][address]', transform: extractTransform },
+  { node: "SpouseCity",  arg: 'veteran[spouses][city]', transform: extractTransform },
+  //{ node: "SpouseCountry",  arg: 'veteran[spouses][country]', transform: extractTransform },
```
Comment (awong-dev) [inline _health-care/_js/_form.js]: Remove commented out chunks?  No reason to commit commented out code if we have revision control.

**Comment by awong-dev** [comment]
LGTM w/ nit.

Feel free to merge after you delete commented out code.

**Pull request merged**